### ISSUE TITLE
Possible problem when port is different by default (5432)

### ISSUE THREAD

**Issue by sasadangelo** [zalando/spilo#617]
Hi all,

I have a Kubernetes file to deploy Spilo. So far everything has worked properly. However, I need to change the port with a value different by default, for this reason I set:
```
PGPORT=5524
```

I deployed the cluster and everything is running. Running the command ```patronictl list``` everything is fine. However, in the Pod logs I see output like this:
```
psql: error: could not connect to server: No such file or directory
	Is the server running locally and accepting
	connections on Unix domain socket "/var/run/postgresql/.s.PGSQL.5432"?
2021-09-03 10:09:05,072 INFO: no action. I am (cfdb-ccdb-0) the leader with the lock
psql: error: could not connect to server: No such file or directory
	Is the server running locally and accepting
	connections on Unix domain socket "/var/run/postgresql/.s.PGSQL.5432"?
psql: error: could not connect to server: No such file or directory
	Is the server running locally and accepting
	connections on Unix domain socket "/var/run/postgresql/.s.PGSQL.5432"?
psql: error: could not connect to server: No such file or directory
	Is the server running locally and accepting
	connections on Unix domain socket "/var/run/postgresql/.s.PGSQL.5432"?
psql: error: could not connect to server: No such file or directory
```

There are still reference to 5432. Looking into the container I found that ```/home/postgres/postgres.yml``` has the following section:
```
kubernetes:
  bypass_api_service: true
  labels:
    application: cfdb-ccdb
  pod_ip: 192.168.140.18
  ports:
  - name: postgresql
    port: 5432
  role_label: role
  scope_label: dbinstance
  use_endpoints: true
```

Basically, it seems that the code that updates the leader still refers to the old port.
Is this a bug?
Should I specify the section above with the right port in the SPILO_CONFIGURATION variable?

**Comment by sasadangelo** [comment]
If I am correct the possible hard-coded value could be here:
https://github.com/zalando/spilo/blob/62d5293df82be487eb14004c77a8a8dd9891655b/postgres-appliance/scripts/configure_spilo.py#L635-L637
Am I right?

**Comment by CyberDem0n** [comment]
Ref: https://github.com/zalando/spilo/issues/469

**Comment by CyberDem0n** [comment]
I just don't understand what is the point in overriding the standard port in containerized infrastructure.

**Comment by sasadangelo** [comment]
I think I found the issue, but it doesn't solve completely the problem. 
My problem was that I overrode the property:
```
unix_socket_directories: /tmp
```

while Spilo, by default, uses ```/var/run/postgresql``` for local connections and there is no way to change this. 
The problem was that I set the port to 5524 so if you look at the logs above PostgreSQL should have a check for socket /var/run/postgresql/.s.PGSQL.5524 instead of /var/run/postgresql/.s.PGSQL.5432.
I thought there was somewhere 5432 hardcoded into the code. The only piece where I found it hardcoded was the code above, but I don't know if it is the origin of the problem.
